src/system_reader.rs

```diff
@@ -0,0 +1,927 @@
+                        self.lst.state = AtLstSymbols;
+                    } else {
+                        // This field has no effect on our handling of the LST.
+                        self.lst.state = AtLstOpenContent;
+                    }
+                } else {
+                    // There's no field name; we're between fields.
+                    self.lst.state = BetweenLstFields;
+                }
+            }
+            ProcessingLstImports => {
+                todo!("Support shared symbol table imports.");
+            }
+            ProcessingLstSymbols => {
+                // We're in the `symbols` list.
+                if let (IonType::String, false) = (ion_type, is_null) {
+                    // If the current value is a non-null string, add its text to the symbol table.
+                    self.load_current_string()?;
+                    // We clone the current string because the user may ask for its value via
+                    // read_string().
+                    self.lst.symbols.push(Some(self.lst.current_string.clone()))
+                } else {
+                    // Non-string values and nulls are treated as symbols with unknown text.
+                    self.lst.symbols.push(None);
+                }
+            }
+            ProcessingLstOpenContent => {
+                // We were in open content before and haven't stepped out yet. Do nothing.
+            }
+        };
+
+        Ok(Some(SystemStreamItem::SymbolTableData(ion_type, is_null)))
+    }
+
+    // Called when the system reader advances to the `imports` field of an LST.
+    fn move_to_lst_imports_field(&mut self, ion_type: IonType) -> IonResult<()> {
+        match ion_type {
+            IonType::Symbol => {
+                // If the `imports` field value is the symbol '$ion_symbol_table', then this is an
+                // LST append.
+                self.load_current_symbol()?;
+                if self
+                    .lst
+                    .current_symbol
+                    .matches(system_symbol_ids::ION_SYMBOL_TABLE, "$ion_symbol_table")
+                {
+                    self.lst.is_append = true;
+                }
+            }
+            IonType::List => {
+                // Once this is supported, this branch will do nothing because the list either
+                // be processed when the user steps into/through it or when they try to skip over
+                // it, not when it's first encountered. For now though, we fail because this
+                // feature is not yet supported.
+                todo!("Support shared symbol table imports.");
+            }
+            _ => {
+                // Non-list, non-symbol values for the `imports` field are ignored.
+            }
+        };
+        Ok(())
+    }
+
+    // Reads the raw reader's current value expecting a symbol. Stores the value in
+    // `self.lst.current_symbol` so it can be returned if the user requests it.
```

Comment:
Is there any reason why you don't use `///` for even these internal methods?  These read like API docs, I am curious why you don't tag them as such even though we won't publish them.

Reply:
I guess I wonder the inverse: why use `///` if they don't get published? I'm happy to switch if there's a good reason. I suppose it would allow IDEs to resolve rustdoc references to other types/methods?

Reply:
The main reason is IDE support, but also you _could_ generate the docs for your own review which is nice.  It definitely costs us nothing and I like the practice of documenting internal APIs as well as we would public ones.